matchzoo/engine/base_model.py

```diff
@@ -21,6 +21,53 @@ class BaseModel(abc.ABC):
     BACKEND_WEIGHTS_FILENAME = 'backend_weights.h5'
     PARAMS_FILENAME = 'params.dill'
 
+    class EvaluateOnCall(keras.callbacks.Callback):
+        def __init__(self, x, y,
+                     metrics: typing.Union[list, str, engine.BaseMetric],
+                     valid_steps=3,
+                     batch_size: int = 32):
+            super().__init__()
+            self._x = x
+            self._y = y
+            self._metrics = metrics
+            self._valid_steps = valid_steps
+            self._batch_size = batch_size
+
+        def on_epoch_end(self, epoch, logs={}):
+            if epoch % self._valid_steps:
+                return logs
+            val_x = self._x
+            val_y = self._y
```

Comment:
why create `val_x` and `val_y`?

Reply:
We should express that these `x` and `y` are for validation, and this is one way. The other way could be naming `self._val_x` at the first place. I am ok with both.